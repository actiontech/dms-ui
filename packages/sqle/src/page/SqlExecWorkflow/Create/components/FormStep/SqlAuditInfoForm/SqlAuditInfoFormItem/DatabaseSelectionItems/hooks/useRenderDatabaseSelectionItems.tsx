import instance from '@actiontech/shared/lib/api/sqle/service/instance';
import { DatabaseSelectionItemProps } from '../../../index.type';
import { useCurrentProject } from '@actiontech/shared/lib/features';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import { BasicButton, BasicToolTip, TypedLink } from '@actiontech/shared';
import { MinusCircleFilled, ProfileSquareFilled } from '@actiontech/icons';
import { useTranslation } from 'react-i18next';
import { FormListFieldData } from 'antd';
import system from '@actiontech/shared/lib/api/sqle/service/system';
import {
  getSystemModuleStatusDbTypeEnum,
  getSystemModuleStatusModuleNameEnum
} from '@actiontech/shared/lib/api/sqle/service/system/index.enum';
import useThemeStyleData from '../../../../../../../../../hooks/useThemeStyleData';
import useCreationMode from '../../../../../../hooks/useCreationMode';
import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { IReduxState } from '../../../../../../../../../store';
import { ROUTE_PATHS } from '@actiontech/shared/lib/data/routePaths';
import { IInstanceTipResV1 } from '@actiontech/shared/lib/api/sqle/service/common';

const useRenderDatabaseSelectionItems = ({
  dbSourceInfoCollection,
  sqlStatementTabActiveKey,
  instanceList
}: Pick<
  DatabaseSelectionItemProps,
  'dbSourceInfoCollection' | 'sqlStatementTabActiveKey'
> & { instanceList?: IInstanceTipResV1[] }) => {
  const { t } = useTranslation();
  const { projectName, projectID } = useCurrentProject();
  const { sqleTheme } = useThemeStyleData();

  const { isCloneMode, isRollbackMode } = useCreationMode();

  const sqlExecWorkflowReduxState = useSelector((state: IReduxState) => {
    return {
      clonedExecWorkflowSqlAuditInfo:
        state.sqlExecWorkflow.clonedExecWorkflowSqlAuditInfo
    };
  });

  useEffect(() => {
    if (isCloneMode || isRollbackMode) {
      sqlExecWorkflowReduxState.clonedExecWorkflowSqlAuditInfo?.databaseInfo?.forEach(
        (database, index) => {
          const key = `${index}`;
          handleInstanceChange(key, database.instanceName);
          handleInstanceSchemaChange(key, database.instanceSchema);
        }
      );
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const updateSchemaList = async (key: string, instanceName: string) => {
    instance
      .getInstanceSchemasV1({
        instance_name: instanceName,
        project_name: projectName
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          dbSourceInfoCollection.set(key, {
            schemaList: res.data.data?.schema_name_list
          });
        }
      })
      .finally(() => {
        dbSourceInfoCollection.set(key, {
          getSchemaLoading: false
        });
      });
  };

  const getSupportedFileModeByInstanceType = (key: string, dbType: string) => {
    system
      .getSystemModuleStatus({
        db_type: dbType as getSystemModuleStatusDbTypeEnum,
        module_name: getSystemModuleStatusModuleNameEnum.execute_sql_file_mode
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          dbSourceInfoCollection.set(key, {
            isSupportFileModeExecuteSql: !!res.data.data?.is_supported
          });
        }
      });
  };

  const updateRuleTemplateNameAndDbType = (
    key: string,
    instanceName: string
  ) => {
    instance
      .getInstanceV2({ instance_name: instanceName, project_name: projectName })
      .then(async (res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          dbSourceInfoCollection.set(key, {
            dbType: res.data.data?.db_type,
            ruleTemplate: res.data.data?.rule_template
          });

          getSupportedFileModeByInstanceType(key, res.data.data?.db_type ?? '');
        }
      });
  };

  const handleInstanceChange = (key: string, instanceName?: string) => {
    if (instanceName) {
      const targetInstance = instanceList?.find(
        (i) => i.instance_name === instanceName
      );
      dbSourceInfoCollection.set(key, {
        instanceName,
        schemaName: undefined,
        getSchemaLoading: true,
        schemaList: [],
        ruleTemplate: undefined,
        dbType: undefined,
        testConnectResult: undefined,
        isSupportFileModeExecuteSql: true,
        enableBackup: targetInstance?.enable_backup ?? false,
        backupMaxRows: targetInstance?.backup_max_rows,
        allowBackup: !!targetInstance?.supported_backup_strategy?.length
      });
      updateSchemaList(key, instanceName);
      updateRuleTemplateNameAndDbType(key, instanceName);
      sqlStatementTabActiveKey.set(key);
    }
  };

  useEffect(() => {
    // 克隆或者回滚时，因为是从store中取instanceName ，此时可能instance list接口还未结束
    // 导致enableBackup被赋予了默认值false，所以需要在instance list接口完成后，重新设置enableBackup
    Object.keys(dbSourceInfoCollection.value).forEach((key) => {
      const targetInstance = instanceList?.find(
        (i) =>
          i.instance_name === dbSourceInfoCollection.value[key].instanceName
      );
      dbSourceInfoCollection.set(key, {
        enableBackup: targetInstance?.enable_backup ?? false,
        backupMaxRows: targetInstance?.backup_max_rows,
        allowBackup: !!targetInstance?.supported_backup_strategy?.length
      });
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [instanceList]);

  const handleInstanceSchemaChange = (key: string, schemaName?: string) => {
    dbSourceInfoCollection.set(key, {
      schemaName
    });
  };

  const getInstanceSchemaOptions = (key: string) => {
    return (
      dbSourceInfoCollection.value?.[key]?.schemaList?.map((item) => {
        return {
          label: item,
          value: item
        };
      }) ?? []
    );
  };

  const getInstanceSchemaLoading = (key: string) => {
    return dbSourceInfoCollection.value?.[key]?.getSchemaLoading;
  };

  const renderRuleTemplateDisplay = (key: string) => {
    const rule = dbSourceInfoCollection.value?.[key]?.ruleTemplate;
    const dbType = dbSourceInfoCollection.value?.[key]?.dbType;

    if (!rule || !dbType) {
      return (
        <BasicButton
          className="data-source-row-rule-template"
          icon={<ProfileSquareFilled width="18" height="18" />}
        />
      );
    }

    const path = rule.is_global_rule_template
      ? ROUTE_PATHS.SQLE.RULE_MANAGEMENT.detail
      : ROUTE_PATHS.SQLE.RULE_TEMPLATE.detail;

    return (
      <BasicToolTip
        title={
          <TypedLink
            to={path}
            params={{
              projectID,
              dbType,
              templateName: rule.name ?? ''
            }}
            target="_blank"
          >
            {t('rule.form.ruleTemplate')}: {rule.name}
          </TypedLink>
        }
      >
        <BasicButton
          className="data-source-row-rule-template"
          icon={
            <ProfileSquareFilled
              width="18"
              height="18"
              color={sqleTheme.icon.execWorkFlow.profileSquareFilled}
            />
          }
        />
      </BasicToolTip>
    );
  };

  const renderDeleteItemButton = (
    fields: FormListFieldData[],
    key: string,
    handleClick: () => void
  ) => {
    const removeItem = () => {
      dbSourceInfoCollection.set(key, undefined);
      sqlStatementTabActiveKey.set(
        Object.keys(dbSourceInfoCollection.value).filter(
          (v) => v !== key
        )?.[0] ?? ''
      );
    };

    return (
      <BasicButton
        className="data-source-row-button data-source-col-delete-button"
        onClick={() => {
          if (fields.length <= 1) {
            return;
          }
          handleClick();
          removeItem();
        }}
        icon={
          fields.length > 1 ? (
            <MinusCircleFilled width="18" height="18" />
          ) : (
            <MinusCircleFilled
              width="18"
              height="18"
              color={sqleTheme.icon.execWorkFlow.minusCircleFilledDisabled}
            />
          )
        }
      />
    );
  };

  const renderAddItemButton = (
    fields: FormListFieldData[],
    handleClick: () => void
  ) => {
    /**
     * 新增时为什么没有同步更新 dbSourceInfoCollection
     * 由于无法获取最新的 key，所以无法直接在这里新增。只能通过选择数据源时更新 dbSourceInfoCollection 数据
     */
    const addAction = () => {
      handleClick();
    };
    return (
      <BasicButton
        onClick={addAction}
        type="primary"
        disabled={fields.length >= 10}
      >
        {t('execWorkflow.create.form.sqlInfo.addInstance')}
      </BasicButton>
    );
  };

  return {
    handleInstanceChange,
    handleInstanceSchemaChange,
    getInstanceSchemaOptions,
    getInstanceSchemaLoading,
    renderRuleTemplateDisplay,
    renderDeleteItemButton,
    renderAddItemButton
  };
};

export default useRenderDatabaseSelectionItems;
