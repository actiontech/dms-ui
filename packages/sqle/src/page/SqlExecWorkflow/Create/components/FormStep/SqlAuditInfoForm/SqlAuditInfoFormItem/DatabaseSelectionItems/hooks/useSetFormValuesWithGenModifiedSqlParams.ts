import { useRequest } from 'ahooks';
import databaseCompareService from '@actiontech/shared/lib/api/sqle/service/database_comparison';
import { IGenDatabaseDiffModifySQLsV1Params } from '@actiontech/shared/lib/api/sqle/service/database_comparison/index.d';
import { decompressFromEncodedURIComponent } from 'lz-string';
import { jsonParse } from '@actiontech/shared/lib/utils/Common';
import { IDatabaseDiffModifySQL } from '@actiontech/shared/lib/api/sqle/service/common';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import { FormInstance } from 'antd';
import {
  SqlAuditInfoFormFields,
  SqlStatementFields
} from '../../../../../../index.type';
import { SAME_SQL_MODE_DEFAULT_FIELD_KEY } from '../../../../../../../Common/SqlStatementFormController/SqlStatementFormItem/index.data';
import { useTypedQuery } from '@actiontech/shared';
import { ROUTE_PATHS } from '@actiontech/shared/lib/data/routePaths';
import { useMemo } from 'react';

type Params = {
  form: FormInstance<SqlAuditInfoFormFields>;
  handleInstanceNameChange?: (name: string) => void;
  handleInstanceChange: (key: string, instanceName?: string) => void;
  handleInstanceSchemaChange: (key: string, schemaName?: string) => void;
  setGetModifiedSQLsPending: (val: boolean) => void;
};

const useSetFormValuesWithGenModifiedSqlParams = ({
  form,
  handleInstanceChange,
  handleInstanceSchemaChange,
  handleInstanceNameChange,
  setGetModifiedSQLsPending
}: Params) => {
  const extraQueries = useTypedQuery();

  const searchParams = useMemo(() => {
    return extraQueries(ROUTE_PATHS.SQLE.SQL_EXEC_WORKFLOW.create);
  }, [extraQueries]);

  const getExistSchemaName = (schemaList: string[], schemaName?: string) => {
    if (schemaName && schemaList.includes(schemaName)) {
      return schemaName;
    }

    return undefined;
  };

  const setFormValues = async (
    comparisonInstanceName: string,
    dbExistingSchemas: string[],
    databaseDiffModifiedSqlInfos: IDatabaseDiffModifySQL[]
  ) => {
    handleInstanceNameChange?.(comparisonInstanceName);

    form.setFieldsValue({
      isSameSqlForAll: databaseDiffModifiedSqlInfos.length === 0,
      databaseInfo: databaseDiffModifiedSqlInfos.map((item) => {
        return {
          instanceName: comparisonInstanceName,
          instanceSchema: item.schema_name
        };
      })
    });

    if (databaseDiffModifiedSqlInfos.length === 1) {
      const schemaName = getExistSchemaName(
        dbExistingSchemas,
        databaseDiffModifiedSqlInfos[0].schema_name
      );

      await handleInstanceChange(
        SAME_SQL_MODE_DEFAULT_FIELD_KEY,
        comparisonInstanceName
      );
      handleInstanceSchemaChange(SAME_SQL_MODE_DEFAULT_FIELD_KEY, schemaName);
      form.setFieldsValue({
        isSameSqlForAll: true,
        databaseInfo: [
          {
            instanceName: comparisonInstanceName,
            instanceSchema: schemaName
          }
        ],
        [SAME_SQL_MODE_DEFAULT_FIELD_KEY]: {
          form_data: databaseDiffModifiedSqlInfos[0].modify_sqls
            ?.flatMap((v) => v.sql_statement)
            ?.join('\n')
        } as SqlStatementFields
      });
    } else {
      for (
        let index = 0;
        index < databaseDiffModifiedSqlInfos.length;
        index++
      ) {
        handleInstanceChange(index.toString(), comparisonInstanceName);

        handleInstanceSchemaChange(
          index.toString(),
          getExistSchemaName(
            dbExistingSchemas,
            databaseDiffModifiedSqlInfos[index].schema_name
          )
        );
      }
      const sqlParams = databaseDiffModifiedSqlInfos.reduce<
        Record<string, SqlStatementFields>
      >((acc, curr, index) => {
        return {
          ...acc,
          [index]: {
            form_data: curr.modify_sqls
              ?.flatMap((v) => v.sql_statement)
              ?.join('\n')
          }
        };
      }, {} as Record<string, SqlStatementFields>);
      form.setFieldsValue({
        isSameSqlForAll: false,
        databaseInfo: databaseDiffModifiedSqlInfos.map((item) => {
          return {
            instanceName: comparisonInstanceName,
            instanceSchema: getExistSchemaName(
              dbExistingSchemas,
              item.schema_name
            )
          };
        }),
        ...sqlParams
      });
    }
  };

  useRequest(
    () => {
      const params = jsonParse<
        IGenDatabaseDiffModifySQLsV1Params & {
          comparisonInstanceName: string;
          dbExistingSchemas: string[];
        }
      >(
        decompressFromEncodedURIComponent(
          searchParams?.gen_modified_sql_params!
        )
      );
      const { comparisonInstanceName, dbExistingSchemas, ...apiParams } =
        params;
      setGetModifiedSQLsPending(true);
      return databaseCompareService
        .genDatabaseDiffModifySQLsV1(apiParams)
        .then((res) => {
          if (res.data.code === ResponseCode.SUCCESS) {
            setFormValues(
              comparisonInstanceName,
              dbExistingSchemas,
              res.data.data ?? []
            );
          }
        })
        .finally(() => {
          setGetModifiedSQLsPending(false);
        });
    },
    {
      ready: !!searchParams && !!searchParams?.gen_modified_sql_params
    }
  );
};

export default useSetFormValuesWithGenModifiedSqlParams;
