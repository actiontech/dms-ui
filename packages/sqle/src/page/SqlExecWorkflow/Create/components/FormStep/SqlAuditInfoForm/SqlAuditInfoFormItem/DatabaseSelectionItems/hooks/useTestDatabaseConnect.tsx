import { BasicToolTips } from '@actiontech/shared';
import instance from '@actiontech/shared/lib/api/sqle/service/instance';
import { TestConnectResultStyleWrapper } from '@actiontech/shared/lib/components/TestDatabaseConnectButton/style';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import { useCurrentProject } from '@actiontech/shared/lib/global';
import { useBoolean } from 'ahooks';
import { useCallback, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { DatabaseSelectionItemProps } from '../../../index.type';
import { IBatchCheckInstanceIsConnectableByNameParams } from '@actiontech/shared/lib/api/sqle/service/instance/index.d';
import { DatabaseSelectionFields } from '../index.type';
import { CloseCircleOutlined, CheckCircleOutlined } from '@actiontech/icons';

const useTestDatabaseConnect = ({
  databaseInfo,
  instanceTestConnectResults
}: {
  databaseInfo: Array<DatabaseSelectionFields>;
  instanceTestConnectResults: DatabaseSelectionItemProps['instanceTestConnectResults'];
}) => {
  const { t } = useTranslation();
  const { projectName } = useCurrentProject();
  const [testLoading, { setTrue: testStart, setFalse: testFinish }] =
    useBoolean();

  const disabledTestConnect = useMemo(
    () =>
      databaseInfo?.map((v) => v?.instanceName)?.filter((v) => !!v).length ===
      0,
    [databaseInfo]
  );

  const testDatabaseConnect = useCallback(async () => {
    const instances =
      (databaseInfo ?? [])
        .map((v) => ({
          name: v?.instanceName
        }))
        ?.filter((v) => !!v.name) ?? [];

    if (instances.length === 0) {
      return;
    }

    const params: IBatchCheckInstanceIsConnectableByNameParams = {
      instances: instances,
      project_name: projectName
    };

    testStart();
    instance
      .batchCheckInstanceIsConnectableByName(params)
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          instanceTestConnectResults.set(res.data.data ?? []);
        }
      })
      .finally(() => {
        testFinish();
      });
  }, [
    databaseInfo,
    instanceTestConnectResults,
    projectName,
    testFinish,
    testStart
  ]);

  const renderTestDatabasesConnectInfo = useCallback(
    (name: string) => {
      const result = instanceTestConnectResults.value?.find(
        (v) => v.instance_name === name
      );
      if (!result) {
        return null;
      }

      return (
        <TestConnectResultStyleWrapper
          success={!!result.is_instance_connectable}
        >
          {!!result.is_instance_connectable && (
            <TestConnectResultStyleWrapper success>
              <CheckCircleOutlined className="custom-icon" />
              {t('common.testDatabaseConnectButton.testSuccess')}
            </TestConnectResultStyleWrapper>
          )}
          {!result.is_instance_connectable && (
            <BasicToolTips title={result.connect_error_message}>
              <TestConnectResultStyleWrapper success={false}>
                <CloseCircleOutlined className="custom-icon" />
                {t('common.testDatabaseConnectButton.testFailed')}
              </TestConnectResultStyleWrapper>
            </BasicToolTips>
          )}
        </TestConnectResultStyleWrapper>
      );
    },
    [instanceTestConnectResults.value, t]
  );

  return {
    testDatabaseConnect,
    testLoading,
    renderTestDatabasesConnectInfo,
    disabledTestConnect
  };
};

export default useTestDatabaseConnect;
