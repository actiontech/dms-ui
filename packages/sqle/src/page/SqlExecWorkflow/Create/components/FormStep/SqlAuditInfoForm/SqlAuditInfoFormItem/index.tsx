import { Form } from 'antd';
import { SqlAuditInfoFormItemProps } from '../index.type';
import DatabaseSelectionItem from './DatabaseSelectionItems';
import { FormItemLabel } from '@actiontech/shared/lib/components/FormCom';
import { useTranslation } from 'react-i18next';
import { BasicSwitch } from '@actiontech/shared';
import { useEffect, useMemo } from 'react';
import {
  CreateWorkflowDatabaseInfo,
  SqlAuditInfoFormFields
} from '../../../../index.type';
import SqlStatementFormController from '../../../../../Common/SqlStatementFormController';
import { SAME_SQL_MODE_DEFAULT_FIELD_KEY } from '../../../../../Common/SqlStatementFormController/SqlStatementFormItem/index.data';
import { RingPieFilled } from '@actiontech/icons';
import { CommonIconStyleWrapper } from '@actiontech/shared/lib/styleWrapper/element';
import { forwardRef } from 'react';

const SqlAuditInfoFormItem = forwardRef<HTMLElement, SqlAuditInfoFormItemProps>(
  ({ auditAction, handleInstanceNameChange, ...sharedStepDetail }, ref) => {
    const { t } = useTranslation();
    const form = Form.useFormInstance<SqlAuditInfoFormFields>();

    const isSameSqlForAll = Form.useWatch('isSameSqlForAll', form);

    const databaseInfo: CreateWorkflowDatabaseInfo = useMemo(() => {
      return Object.keys(sharedStepDetail.dbSourceInfoCollection.value)
        .map((key) => {
          return {
            key,
            instanceName:
              sharedStepDetail.dbSourceInfoCollection.value?.[key]
                ?.instanceName,
            schemaName:
              sharedStepDetail.dbSourceInfoCollection.value?.[key]?.schemaName,
            enableBackup:
              sharedStepDetail.dbSourceInfoCollection.value?.[key]
                ?.enableBackup,
            allowBackup:
              sharedStepDetail.dbSourceInfoCollection.value?.[key]?.allowBackup,
            backupMaxRows:
              sharedStepDetail.dbSourceInfoCollection.value?.[key]
                ?.backupMaxRows
          };
        })
        .filter((v) => !!v.instanceName);
    }, [sharedStepDetail.dbSourceInfoCollection]);

    const isSupportFileModeExecuteSqlRecord: Record<string, boolean> =
      useMemo(() => {
        if (isSameSqlForAll) {
          // 相同 SQL 模式下，判断所有数据源是否都支持。虽然 相同 SQL 模式下所有数据源类型会相同，但还是做下校验
          return {
            [SAME_SQL_MODE_DEFAULT_FIELD_KEY]: Object.keys(
              sharedStepDetail.dbSourceInfoCollection.value
            ).every((key) => {
              return !!sharedStepDetail.dbSourceInfoCollection?.value?.[key]
                ?.isSupportFileModeExecuteSql;
            })
          };
        }
        //不同 SQL 模式下，每个数据源对应各自当前数据源类型是否支持文件上线模式
        return Object.keys(
          sharedStepDetail.dbSourceInfoCollection.value
        ).reduce((acc, key) => {
          return {
            ...acc,
            [key]:
              !!sharedStepDetail.dbSourceInfoCollection?.value?.[key]
                ?.isSupportFileModeExecuteSql
          };
        }, {});
      }, [isSameSqlForAll, sharedStepDetail.dbSourceInfoCollection.value]);

    useEffect(() => {
      const dbTypeSet = new Set(
        Object.keys(sharedStepDetail.dbSourceInfoCollection.value)
          ?.map((key) => {
            return sharedStepDetail.dbSourceInfoCollection.value?.[key].dbType;
          })
          .filter((v) => !!v)
      );
      if (dbTypeSet.size > 1) {
        form.setFieldValue('isSameSqlForAll', false);
        sharedStepDetail.isDisabledForDifferenceSql.set(true);
      } else {
        sharedStepDetail.isDisabledForDifferenceSql.set(false);
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [form, sharedStepDetail.dbSourceInfoCollection.value]);

    return (
      <>
        <section ref={ref}>
          <DatabaseSelectionItem
            {...sharedStepDetail}
            handleInstanceNameChange={handleInstanceNameChange}
          />
        </section>

        <FormItemLabel
          // #if [ce]
          hidden={true}
          // #endif
          name="isSameSqlForAll"
          valuePropName="checked"
          initialValue={true}
          className="has-label-tip"
          label={
            <div className="label-cont-custom">
              <div>
                <CommonIconStyleWrapper className="custom-icon-ellipse">
                  <RingPieFilled />
                </CommonIconStyleWrapper>
                <span>
                  {t('execWorkflow.create.form.sqlInfo.isSameSqlForAll')}
                </span>
              </div>
              <div className="tip-content-box">
                {t(
                  'execWorkflow.create.form.sqlInfo.tipsDataSourceTypeForSameSql'
                )}
              </div>
            </div>
          }
          labelCol={{ span: 22 }}
          wrapperCol={{ span: 2 }}
        >
          <BasicSwitch
            disabled={sharedStepDetail.isDisabledForDifferenceSql.value}
          />
        </FormItemLabel>

        <SqlStatementFormController
          activeKey={sharedStepDetail.sqlStatementTabActiveKey.value}
          onChange={sharedStepDetail.sqlStatementTabActiveKey.set}
          auditAction={auditAction}
          isSameSqlForAll={isSameSqlForAll}
          databaseInfo={databaseInfo}
          isAuditing={sharedStepDetail.isAuditing}
          isSupportFileModeExecuteSqlRecord={isSupportFileModeExecuteSqlRecord}
          isAtFormStep={sharedStepDetail.isAtFormStep}
        />
      </>
    );
  }
);

export default SqlAuditInfoFormItem;
