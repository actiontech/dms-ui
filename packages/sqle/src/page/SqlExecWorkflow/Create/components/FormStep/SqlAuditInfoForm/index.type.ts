import { SharedStepDetails } from '../../../index.type';
import { CreateWorkflowFormStepProps } from '../index.type';

export type SqlAuditInfoFormProps = {
  handleInstanceNameChange?: (name: string) => void;
} & SharedStepDetails &
  Pick<CreateWorkflowFormStepProps, 'auditAction'>;

export type SqlAuditInfoFormItemProps = SqlAuditInfoFormProps;

export type DatabaseSelectionItemProps = Omit<
  SqlAuditInfoFormProps,
  'auditAction' | ''
>;
