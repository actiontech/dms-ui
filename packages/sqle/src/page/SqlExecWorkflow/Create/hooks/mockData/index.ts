import { SharedStepDetails } from '../../index.type';

export const MockSharedStepDetail: SharedStepDetails = {
  isAuditing: {
    value: false,
    set: jest.fn()
  },
  dbSourceInfoCollection: {
    value: { '0': {} },
    set: jest.fn()
  },
  instanceTestConnectResults: {
    value: [],
    set: jest.fn()
  },
  isDisabledForDifferenceSql: {
    value: false,
    set: jest.fn()
  },
  sqlStatementTabActiveKey: {
    value: '1',
    set: jest.fn()
  },
  resetAllSharedData: jest.fn()
};
