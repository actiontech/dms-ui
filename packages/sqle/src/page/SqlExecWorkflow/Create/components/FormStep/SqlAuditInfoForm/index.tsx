import { FormItemSubTitle } from '@actiontech/shared/lib/components/CustomForm';
import { FormAreaBlockStyleWrapper } from '@actiontech/shared/lib/components/CustomForm/style';
import { useTranslation } from 'react-i18next';
import SqlAuditInfoFormItem from './SqlAuditInfoFormItem';
import { SqlAuditInfoFormProps } from './index.type';
import { forwardRef } from 'react';

const SqlAuditInfoForm = forwardRef<HTMLElement, SqlAuditInfoFormProps>(
  (props, ref) => {
    const { t } = useTranslation();
    return (
      <FormAreaBlockStyleWrapper>
        <FormItemSubTitle>
          {t('execWorkflow.create.form.sqlInfo.title')}
        </FormItemSubTitle>
        <SqlAuditInfoFormItem {...props} ref={ref} />
      </FormAreaBlockStyleWrapper>
    );
  }
);

export default SqlAuditInfoForm;
