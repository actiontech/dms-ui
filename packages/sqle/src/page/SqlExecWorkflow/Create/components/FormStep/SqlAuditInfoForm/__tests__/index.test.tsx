import { shallow } from 'enzyme';
import SqlAuditInfoForm from '..';
import toJson from 'enzyme-to-json';
import { MockSharedStepDetail } from '../../../../hooks/mockData';

describe('test SqlAuditInfoForm', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <SqlAuditInfoForm
        handleInstanceNameChange={jest.fn()}
        auditAction={jest.fn()}
        {...MockSharedStepDetail}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
