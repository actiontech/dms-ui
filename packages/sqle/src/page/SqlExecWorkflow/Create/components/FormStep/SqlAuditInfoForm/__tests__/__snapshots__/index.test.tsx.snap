// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SqlAuditInfoForm should match snapshot 1`] = `
<Styled(section)>
  <FormItemSubTitle>
    审核SQL语句信息
  </FormItemSubTitle>
  <ForwardRef
    auditAction={[MockFunction]}
    dbSourceInfoCollection={
      {
        "set": [MockFunction],
        "value": {
          "0": {},
        },
      }
    }
    handleInstanceNameChange={[MockFunction]}
    instanceTestConnectResults={
      {
        "set": [MockFunction],
        "value": [],
      }
    }
    isAuditing={
      {
        "set": [MockFunction],
        "value": false,
      }
    }
    isDisabledForDifferenceSql={
      {
        "set": [MockFunction],
        "value": false,
      }
    }
    resetAllSharedData={[MockFunction]}
    sqlStatementTabActiveKey={
      {
        "set": [MockFunction],
        "value": "1",
      }
    }
  />
</Styled(section)>
`;
